napari/components/_viewer_key_bindings.py

```diff
@@ -59,15 +58,20 @@ def toggle_ndisplay(viewer: Viewer):
 @register_viewer_action(trans._("Toggle theme."))
 def toggle_theme(viewer: Viewer):
     """Toggle theme for viewer"""
-    settings = get_settings()
     themes = available_themes()
-    current_theme = settings.appearance.theme
+    current_theme = viewer.theme
+    # Check what the system theme is, to toggle properly
+    if current_theme == 'system':
+        current_theme = get_system_theme()
     idx = themes.index(current_theme)
     idx += 1
-    if idx == len(themes):
+    # Don't toggle to system, just among actual themes
+    if themes[idx] == 'system':
+        idx += 1
+    if idx >= len(themes):
```

Comment:
I think that this if order should be reverted.

Reply:
You mean skipping `system`? I think skipping it in the toggle is desired behavior, see my answer to Lorenzo: https://github.com/napari/napari/pull/5143#discussion_r980912729
Or do you mean the order of the if statements? My logic was as follows:
Want to skip `system`, but `system` isn't always last. So if we're Dark (say idx 0) and we hit the toggle, we increment idx to 1, check if `system`. If `system`, increment again, to skip it, so idx is then 2. Then we check if we're still within the length of themes, but if the last theme was last or the last theme was *next to last* and `system` was last, then we wrap around to 0.

Reply:
Ok. I see that I'm not writing everything. in line 67 there is the increment of `idx`. After this, you check `themes[idx]`. And in this place, you could get the index error. 
So before this check also should be verified. 

During the test you do not hit the problem because `"system"` theme is at the last position:

https://github.com/napari/napari/blob/8bb7e78f8fa9ec05fa82302c9e27254a85c1cc11/napari/utils/theme.py#L274

But if you change the theme to the `system` in the settings of napari and then call this action then you should expect an exception. 

But of course, the next if is required as there will be (for the current implementation of themes order) need to change it to zero.

Reply:
As always you're 100% correct.
I've fixed it, but I can't figure how to do it without nesting another `if`.
So now I increment, check vs. `len(themes) then check vs. `system`
But if `system`, I increment again, to skip `system` which requires checking `len(themes)` again.
Doesn't seem pythonic? 🤷‍♂️

Reply:
I add a shorter version, but I'm not sure if it is more pythonic.

Reply:
Should be resolved, went with the modulo method you suggested.